src/main/java/mate/academy/hibernate/relations/dao/impl/MovieDaoImpl.java

```diff
@@ -12,11 +15,34 @@ public MovieDaoImpl(SessionFactory sessionFactory) {
 
     @Override
     public Movie add(Movie movie) {
-        return null;
+        Session session = null;
+        Transaction transaction = null;
+        try {
+            session = factory.openSession();
+            transaction = session.beginTransaction();
+            session.persist(movie);
+            transaction.commit();
+        } catch (RuntimeException e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException(String.format(
+                    "Cannot save movie:%s to DB", movie.getTitle()), e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+        return movie;
     }
 
     @Override
     public Optional<Movie> get(Long id) {
-        return null;
+        try (Session session = factory.openSession()) {
+            return Optional.ofNullable(session.get(Movie.class, id));
```

Comment:
Use try with resources to ensure that the session is closed properly. This is a Read operation, and it doesn't require a separate transaction block, so try with resources is more appropriate here.